### DISCUSSION TITLE
DC gain of an integrating (unstable) system

### DISCUSSION THREAD

**Discussion by jamestjsp** [ilayn/harold#89] [Ideas] [answered]
The definition of gain for integrator is the slope of the curve for a unit step input.
Algorithm:
1. Discretize the system to the time unit of gain.
2. Generate unit step response for a predefined length of time (time to steady state).
3.  Report the difference of last two step response coefficient as dcgain.

**Comment by ilayn** [comment, marked as answer]
I have never heard that definition for DC gain. Doesn't sound like correct either.

**Reply by jamestjsp** [reply]
As we know the dc gain of an integrator is +inf or -inf, but from a controller point of view there will be a time window to control the process, isn't it? See the case of a level controller of a tank with one in flow and one out flow. The level is an integrator. Tuning of a pid controller for such a system need this ramp rate (in process industry we call it ramp gain).

**Reply by ilayn** [reply]
Yes that's clear but that's not DC gain. But what you are asking is DC factor or sometimes called DC equivalent. You can find it by getting the DC gain of `s*G(s)`.

**Reply by jamestjsp** [reply]
Yes, I have visualize it from  impulse response plot! Thank you for your kindness!
